app/controllers/transcribe_controller.rb

```diff
@@ -123,113 +123,126 @@ def needs_review
+    save_to_needs_review = params[:save_to_needs_review] || params[:done_to_needs_review]
+    save_to_transcribed = params[:save_to_transcribed] || params[:done_to_transcribed]
+    approve_to_transcribed = params[:approve_to_transcribed]
+
+    if params['save'] || save_to_incomplete || save_to_needs_review || save_to_transcribed || approve_to_transcribed
       message = log_transcript_attempt
-      #leave the status alone if it's needs review, but otherwise set it to transcribed
 
-      if params['save_to_incomplete'] && params[:page]['needs_review'] != '1' 
+      # leave the status alone if it's needs review, but otherwise set it to transcribed
+      if save_to_incomplete && params[:page]['needs_review'] != '1'
         @page.status = Page::STATUS_INCOMPLETE
-      elsif params['save_to_needs_review']
+      elsif params[:done_to_needs_review] && @page.work.collection.review_workflow
         @page.status = Page::STATUS_NEEDS_REVIEW
-      elsif (params['save_to_transcribed'] && params[:page]['needs_review'] != '1') || params['approve_to_transcribed']
+      elsif save_to_needs_review
+        @page.status = params[:page]['needs_review'] == '1' ? Page::STATUS_NEEDS_REVIEW : Page::STATUS_INCOMPLETE
+      elsif (save_to_transcribed && params[:page]['needs_review'] != '1') || approve_to_transcribed
         @page.status = Page::STATUS_TRANSCRIBED
       else
         # old code; possibly dead
-        unless @page.status == Page::STATUS_NEEDS_REVIEW
-          @page.status = Page::STATUS_TRANSCRIBED
-        end
+        @page.status = Page::STATUS_TRANSCRIBED unless @page.status == Page::STATUS_NEEDS_REVIEW
       end
 
       begin
         if @page.save
-          if @page.field_based && !table_cells.empty?
-            @page.replace_table_cells(table_cells)
-          end
+          @page.replace_table_cells(table_cells) if @page.field_based && !table_cells.empty?
 
           log_transcript_success
           flash[:notice] = t('.saved_notice')
+
           if @page.work.ocr_correction
             record_deed(DeedType::OCR_CORRECTED)
-          else
-            if @page.source_text_previously_changed?
-              record_transcription_deed
-            end
+          elsif @page.source_text_previously_changed?
+            record_transcription_deed
           end
-          #don't reset subjects if they're disabled
-          unless @page.collection.subjects_disabled || (@page.source_text.include?("[[") == false)
+
+          # don't reset subjects if they're disabled
+          unless @page.collection.subjects_disabled || (@page.source_text.include?('[[') == false)
             # use the new links to blank the graphs
             @page.clear_article_graphs
 
             new_link_count = @page.page_article_links.where(text_type: 'transcription').count
             logger.debug("DEBUG old_link_count=#{old_link_count}, new_link_count=#{new_link_count}")
```

Comment:
Let's remove this debug statement; we haven't used it for a decade.  (If that makes the `_link_count` variables redundant, we can remove them too.)